src/core/Akka/Pattern/CircuitBreaker.cs

```diff
@@ -274,7 +276,7 @@ private void Transition(AtomicState fromState, AtomicState toState)
             if (SwapState(fromState, toState))
             {
                 Debug.WriteLine("Successful transition from {0} to {1}", fromState, toState);
-                toState.Enter();
+                toState.Enter(fromState.LastCaughtException);
```

Comment:
Do we need to pass the old exception during the transition? Shouldn't the new transtion start with a clean slate?

Reply:
No, the last caught exception need to be passed from one state to another, for example, exception caught in open state needs to be passed to the closed state, so that failing fast can report the exception that caused the state to close, same thing with closed to half open.

Reply:
I guess another way of doing this by making `OnFail` and `OnSuccess` callback and store/clear the last caught exception in the circuit breaker class instead, do you think that'll be cleaner?

Reply:
> I guess another way of doing this by making `OnFail` and `OnSuccess` callback and store/clear the last caught exception in the circuit breaker class instead, do you think that'll be cleaner?

Yes, I think that's a cleaner way of doing this. Let's do that.